stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/SendMessageListenerState.kt

```diff
@@ -88,16 +90,11 @@ internal class SendMessageListenerImpl(
      *
      * @return [Message] Updated message.
      */
-    private suspend fun handleSendMessageFail(
+    private fun handleSendMessageFail(
         logic: LogicRegistry,
         message: Message,
-        error: ChatError
+        error: ChatError,
     ) {
-        // Don't update latest message with this id if it is already synced.
-        val latestUpdatedMessage = repos.selectMessage(message.id) ?: message
-        if (latestUpdatedMessage.syncStatus == SyncStatus.COMPLETED) {
-            return
-        }
```

Comment:
Shouldn't we try to look for this message within the State? 🤔

Reply:
I thought about it too. But I personally don't like this check... in my way to see, if a message was successfully sent to the backend it should be inserted/updated in the database and state. This operation can happen more than once production the same results, so I don't don't the check is needed. 

What do you think?

Reply:
But what happens then if the `Offline` plugin is not plugged in?

It seems like `repos.selectMessage(message.id)` will always be returning `null` and we always gonna be using the `message` which is coming from params as a fallback.

Reply:
Yes, it will. You know, we may run in a situation where added side effects to `ChannelLogic` and `ThreadLogic` run multiple times and that's probably not what we want. Let me add the check for both `SendMessageListenerState` and `SendMessageListenerDatabase`